src/main/java/com/github/fppt/jedismock/operations/scripting/LuaRedisCallback.java

```diff
@@ -0,0 +1,199 @@
+    }
+
+    public LuaValue pcall(final String operationName) {
+        return pcallWrap(() -> call(operationName));
+    }
+
+    public LuaValue pcall(final String operationName, final String rawArg1) {
+        return pcallWrap(() -> call(operationName, rawArg1));
+    }
+
+    public LuaValue pcall(final String operationName, final String rawArg1, final String rawArg2) {
+        return pcallWrap(() -> call(operationName, rawArg1, rawArg2));
+    }
+
+    public LuaValue pcall(final String operationName, final String rawArg1, final String rawArg2, final String rawArg3) {
+        return pcallWrap(() -> call(operationName, rawArg1, rawArg2, rawArg3));
+    }
+
+
+    private static LuaValue pcallWrap(Callable<LuaValue> call) {
+        try {
+            return call.call();
+        } catch (final Exception e) {
+            LuaTable errorTable = new LuaTable();
+            errorTable.set(LuaValue.valueOf("err"), LuaValue.valueOf(e.getMessage()));
+            return errorTable;
+        }
+    }
+
+    private LuaValue execute(final String operationName, final List<Slice> args) {
+        final RedisOperation operation = CommandFactory.buildOperation(operationName.toLowerCase(), true, state, args);
+        if (operation != null) {
+            throwOnUnsupported(operation);
+            return processResultSlice(operationName, operation.execute());
+        }
+        throw new RuntimeException("Operation not implemented!");
+    }
+
+    private static LuaValue processResultSlice(String operationName, Slice result) {
+        if (result != null) {
+            byte[] data = Response.clientResponse(operationName, result).data();
+            return process(getRedisInputStream(data));
+        } else {
+            return LuaValue.NONE;
+        }
+    }
+
+    private static void throwOnUnsupported(RedisOperation operation) {
+        if (operation.getClass().equals(Eval.class)) {
+            throw new RuntimeException("This Redis command is not allowed from scripts");
+        }
+    }
+
+    private static RedisInputStream getRedisInputStream(byte[] data) {
+        return Stream.of(data)
+                .map(ByteArrayInputStream::new)
+                .map(RedisInputStream::new)
+                .collect(Collectors.toList()).get(0);
+    }
+
+    private static LuaValue process(final RedisInputStream is) {
+        byte b = is.readByte();
+        switch (b) {
+            case PLUS_BYTE:
+                return LuaValue.valueOf(processStatusCodeReply(is));
+            case DOLLAR_BYTE:
+                return LuaValue.valueOf(processBulkReply(is));
+            case ASTERISK_BYTE:
+                return embedLuaListToValue(processMultiBulkReply(is));
+            case COLON_BYTE:
+                return LuaValue.valueOf(processInteger(is));
+            case MINUS_BYTE:
+                try {
+                    Method method = Protocol.class.getDeclaredMethod("processError", RedisInputStream.class);
```

Comment:
There is no need in using this method (moreover, via reflection) as no other outcome besides JedisDataException can occure in this context